### PR TITLE
test(infinite-scroll): scroll threshold

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): Tests

## What is the current behavior?

There are no integration tests that validate infinite scroll fires correctly.

## What is the new behavior?

Tests have been added to cover that infinite scroll correctly fires events to load more items (once scrolling to the bottom).

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/utils/test/utils.ts [hunk 1]
```diff
@@ -185,3 +185,20 @@ export const checkModeClasses = async (el: E2EElement, globalMode: string) => {
...
+ * @param event The name of the event.
+ * @param timeout The amount of time to wait for the event to fire (defaults 5000ms)
+ * @returns Resolves when the event fires or the timeout is reached.
+ */
+export const waitForEvent = async (page: E2EPage, event: string, timeout = 5000) => {
+  return Promise.race([
+    page.evaluate(
+      event => new Promise(resolve => document.addEventListener(event, resolve, { once: true })),
+      event
+    ),
```
Comment (liamdebeasi) [inline core/src/utils/test/utils.ts:201]: This feature already exists in the Stencil/Puppeteer testing library. You can do `const ev = await page.spyOnEvent(...);` and then do `await ev.next();`.

Comment (liamdebeasi) [reply core/src/utils/test/utils.ts:201]: Also it looks like `listenForEvent` is redundant too if we use `spyOnEvent`

File: core/src/components/infinite-scroll/test/basic/e2e.ts [hunk 2]
```diff
@@ -1,10 +1,44 @@
...
+  describe('when enabled', () => {
+
+    it('should load more items when scrolled to the bottom', async () => {
+      const page = await newE2EPage({
+        url: '/src/components/infinite-scroll/test/basic?ionic:_testing=true'
+      });
+
+      await scrollIonContentPage(page);
+
+      const items = await page.findAll('ion-item');
```
Comment (liamdebeasi) [inline core/src/components/infinite-scroll/test/basic/e2e.ts:37]: Can we add a check before the scrollIonContentPage that verifies the number of items prior to infinite scroll appending more?

File: core/src/utils/test/utils.ts (hunk 1 above)
Comment (sean-perkins) [reply core/src/utils/test/utils.ts]: Ahh... that works flawlessly. Thank you! I created a task in our tech debt epic to update existing `listenForEvent` usages over to `spyOnEvent`.

**Review by amandaesmith3 — APPROVED** [review]

**Pull request merged**